src/Plugin.js

```diff
@@ -1,5 +1,4 @@
 /* eslint-disable class-methods-use-this, no-unused-vars */
-import datalayer from './datalayer';
```

Comment:
This is not the datalayer instance wich i would expect

Reply:
So which one did you expect here? We (intentionally) have exactly one single Datalayer instance that gets created in the main module and is exported as default export.

Reply:
This import resolves into a new instance. Have a look: https://github.com/ryx/datalayer.js/blob/3a69035e180ca2e8b6f8533212a409db53c8fc08/src/datalayer.js#L228

Reply:
We could bypass this as well with a singleton pattern of course.

Reply:
But shouldn't this only happen exactly once when the module is loaded by node.js for the very first time? If you import the module more than once, you should get the original reference again due to the way module lookup/caching works.

Reply:
Sure... thats what I meant with singleton. But then we should get rid of the first export... 
https://github.com/ryx/datalayer.js/blob/3a69035e180ca2e8b6f8533212a409db53c8fc08/src/datalayer.js#L24
Some developers might get a little bit confused and import *Datalayer* (class) and not the instance.

Reply:
Because if you import just the Datalayer and make a new instance in the bootstrap, the plugins are loosing the functionality to logg.

Reply:
True. That's a rather nasty part of the implementation and is (up until now) needed for testing situations only. Although I think a developer should be aware of the difference between importing and instantiating a class versus importing an instance and using it directly. 

But you're right - it is confusing for anyone new to the code. But the right way to solve this would be to make sure that we use the one global instance in all situations. Creating two-way references between Plugin and Datalayer seems like the wrong approach, if we agree that there should be only one Datalayer instance anyway.

Reply:
Currently you can't import the the datalayer instance via
`import {
  Datalayer,
  attribution,
  annotations,
  metadata,
  methodQueue,
  logger,
  // mutations,
} from 'datalayerjs';`

Reply:
> But if the class export is only used for testing reasons, then we shouldn't export it globally.
With the current implementation of the tests it is almost inevitable to use the `Datalayer` class export, as it is used to reset the datalayer state between two test cases. Alternatively we would have to add something like a `reset` method to the Datalayer, which would allow to cleanup everything after each run.

This is also a serious problem in the internal implementation of the bootstrap module in the code at GKH. If we change it here, then you will likely have to refactor the whole bootstrapping process.

Reply:
> Mixing default and named exports is kind of bad habit

I think that highly depends on the usecase. The notation `import datalayer, { Plugin } from 'datalayerjs';` makes pretty clear what this is all about and that we import a singleton and a class here. 

I'd totally agree that the imports of the extensions aren't very elegant, though. But keeping them in separate paths (as common with node `require('package/path/to/file')` syntax) sadly is not possible with ES6 imports.

Reply:
I'll merge this now so your problems are solved for the moment. For the 0.1.x version we will re-think and improve the singleton-based vs. class-based instantiation issue so this becomes less confusing.